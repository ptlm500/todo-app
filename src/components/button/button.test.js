import React from 'react';
import renderer from 'react-test-renderer';
import {
  PrimaryButton,
  SecondaryButton
} from './Button';
import 'jest-styled-components';

describe('PrimaryButton', () => {
  it('renders a Primary button', () => {
    const tree = renderer.create(<PrimaryButton />);
    expect(tree).toMatchSnapshot();
  });

  it('renders a disabled Primary button', () => {
    expect(renderer.create(<PrimaryButton disabled />).toJSON()).toMatchSnapshot();
  });

  it('renders a Primary danger button', () => {
    const tree = renderer.create(<PrimaryButton danger />);
    expect(tree).toMatchSnapshot();
  });
});

describe('SecondaryButton', () => {
  it('renders a SecondaryButton button', () => {
    const tree = renderer.create(<SecondaryButton />);
    expect(tree).toMatchSnapshot();
  });

  it('renders a SecondaryButton Primary button', () => {
    expect(renderer.create(<SecondaryButton disabled />).toJSON()).toMatchSnapshot();
  });

  it('renders a SecondaryButton danger button', () => {
    const tree = renderer.create(<SecondaryButton danger />);
    expect(tree).toMatchSnapshot();
  });
});
