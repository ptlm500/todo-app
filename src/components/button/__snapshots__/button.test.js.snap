// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders a Primary button 1`] = `
.c0 {
  background: #3F51B5;
  border-radius: 0.5rem;
  border: 2px solid #3F51B5;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c0:hover {
  background: #C5CAE9;
}

.c0:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #757575;
}

<button
  className="c0"
/>
`;

exports[`PrimaryButton renders a Primary danger button 1`] = `
.c0 {
  background: #D32F2F;
  border-radius: 0.5rem;
  border: 2px solid #D32F2F;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c0:hover {
  background: #F44336;
}

.c0:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #757575;
}

<button
  className="c0"
/>
`;

exports[`PrimaryButton renders a disabled Primary button 1`] = `
.c0 {
  background: #3F51B5;
  border-radius: 0.5rem;
  border: 2px solid #3F51B5;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c0:hover {
  background: #C5CAE9;
}

.c0:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #757575;
}

<button
  className="c0"
  disabled={true}
/>
`;

exports[`SecondaryButton renders a SecondaryButton Primary button 1`] = `
.c0 {
  background: #3F51B5;
  border-radius: 0.5rem;
  border: 2px solid #3F51B5;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  background: #FFFFFF;
  border: 2px solid #3F51B5;
  color: #3F51B5;
}

.c0:hover {
  background: #C5CAE9;
}

.c0:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #757575;
}

.c0:hover {
  background: #3F51B5;
  color: #FFFFFF;
}

.c0:disabled {
  background: #FFFFFF;
  border: 2px solid #BDBDBD;
  color: #BDBDBD;
  cursor: auto;
}

<button
  className="sc-AxjAm c0"
  disabled={true}
/>
`;

exports[`SecondaryButton renders a SecondaryButton button 1`] = `
.c0 {
  background: #3F51B5;
  border-radius: 0.5rem;
  border: 2px solid #3F51B5;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  background: #FFFFFF;
  border: 2px solid #3F51B5;
  color: #3F51B5;
}

.c0:hover {
  background: #C5CAE9;
}

.c0:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #757575;
}

.c0:hover {
  background: #3F51B5;
  color: #FFFFFF;
}

.c0:disabled {
  background: #FFFFFF;
  border: 2px solid #BDBDBD;
  color: #BDBDBD;
  cursor: auto;
}

<button
  className="sc-AxjAm c0"
/>
`;

exports[`SecondaryButton renders a SecondaryButton danger button 1`] = `
.c0 {
  background: #D32F2F;
  border-radius: 0.5rem;
  border: 2px solid #D32F2F;
  cursor: pointer;
  color: #FFFFFF;
  height: 2rem;
  padding: 0.25rem 1rem;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  background: #FFFFFF;
  border: 2px solid #D32F2F;
  color: #D32F2F;
}

.c0:hover {
  background: #F44336;
}

.c0:disabled {
  border: 2px solid #BDBDBD;
  background: #BDBDBD;
  color: #757575;
}

.c0:hover {
  background: #D32F2F;
  color: #FFFFFF;
}

.c0:disabled {
  background: #FFFFFF;
  border: 2px solid #BDBDBD;
  color: #BDBDBD;
  cursor: auto;
}

<button
  className="sc-AxjAm c0"
/>
`;
